pkg/platform/kube/functionres/lazy.go

```diff
@@ -1579,7 +1582,8 @@ func (lc *lazyClient) generateCronTriggerCronJobSpec(functionLabels labels.Set,
 		},
 	}
 
-	// set concurrency policy if given
+	// set concurrency policy if given (default to forbid - to protect the user from overdose of cron jobs)
+	attributes.ConcurrencyPolicy = string(batchv1beta1.ForbidConcurrent)
```

Comment:
1. Didn't we say that tightening the cleanup logic will allow us to keep the concurrency policy to allow?
2. You're setting `attributes.ConcurrencyPolicy` and right afterwards check if it's empty :|

Reply:
1. I tried tightening up the cleanup logic (in several ways, e.g. removing stale pending pods), but I found out that using multiple short-interval cron-jobs and allowing concurrency may end up creating too many pods, which will cause too much stress on the system